src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/DynamicFlags.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    /// <remarks>
+    /// WARNING: Mutable struct.
+    /// </remarks>
+    internal struct DynamicFlags
```

Comment:
I've changed it to a class hierarchy with a special subtype for the always-false iterator.  (And I went with "Iterator" since "Enumerator" has strong connotations in .NET.)